gemma.cc

```diff
@@ -569,6 +555,31 @@ void Transformer(int token, size_t pos,
                  kModelDim);
 }
 
+template <class TConfig>
+void RangeChecks(size_t& max_tokens, size_t& max_generated_tokens,
+                 size_t& prompt_size) {
+  if (max_tokens > TConfig::kSeqLen) {
+    fprintf(stderr, "WARNING: max_tokens %zu > kSeqLen %d, truncating.\n",
```

Comment:
Beyond the scope of this PR but one ambitious TODO (https://github.com/google/gemma.cpp/wiki/Notes-on-Tasks-to-be-Done-(03%E2%80%9015%E2%80%902024)) is to unify error handling as return values beyond printing and/or immediately exiting.

Ideally failure/warning states can be passed back to application code to handle in a that's appropriate for the use case.

Reply:
hm, not sure I've seen anything more fine-grained than "user error" (like here) or "generic fail". Is there an example of something we could recover from?

Reply:
Some examples off the top of my head:

- In an application / API service, loading the model file fails you want to give the user feedback (eg an onscreen message or popup) and give the user a chance to retry with a different file without exiting the application.
- In a bulk inference use case, the frontend/application may want to take all the inferences that failed and retry them using a different model or different configuration.

Reply:
These situations come into play more in library use cases moreso than the interactive `run.cc` app, but then most productive use cases will probably be in this category.